akka-stream/src/main/scala/akka/stream/impl/io/IOSources.scala

```diff
@@ -20,7 +20,7 @@ import scala.concurrent.{ Future, Promise }
 private[akka] final class SynchronousFileSource(f: File, chunkSize: Int, val attributes: Attributes, shape: SourceShape[ByteString])
   extends SourceModule[ByteString, Future[Long]](shape) {
   override def create(context: MaterializationContext) = {
-    val mat = ActorMaterializer.downcast(context.materializer)
+    val mat = ActorMaterializer.downcast(context.materializer) //FIXME why does this need to rely on a concrete Materializer impl?
```

Comment:
Right, `AsyncStage` should be enough. I think this impl happened before we had async stage... not sure. Please leave a readme (or maybe better, ticket)